group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -434,22 +437,36 @@ public void setSubscriptionMetadata(
     public Map<String, TopicMetadata> computeSubscriptionMetadata(
         ConsumerGroupMember oldMember,
         ConsumerGroupMember newMember,
-        TopicsImage topicsImage
+        TopicsImage topicsImage,
+        ClusterImage clusterImage
     ) {
         // Copy and update the current subscriptions.
         Map<String, Integer> subscribedTopicNames = new HashMap<>(this.subscribedTopicNames);
         maybeUpdateSubscribedTopicNames(subscribedTopicNames, oldMember, newMember);
 
         // Create the topic metadata for each subscribed topic.
         Map<String, TopicMetadata> newSubscriptionMetadata = new HashMap<>(subscribedTopicNames.size());
+
         subscribedTopicNames.forEach((topicName, count) -> {
             TopicImage topicImage = topicsImage.getTopic(topicName);
             if (topicImage != null) {
+                Map<Integer, Set<String>> partitionRackInfo = new HashMap<>();
+
+                topicImage.partitions().forEach((partition, partitionRegistration) -> {
+                    Set<String> racks = new HashSet<>();
+                    for (int replica : Objects.requireNonNull(partitionRegistration.replicas)) {
```

Comment:
Can `replicas` be null in the image? `requireNonNull` will throw an exception if `replicas` is null. It would be better to handle it gracefully if it can really happen.

Reply:
I don't think so, I just kept it as a precautionary measure

Reply:
Let's remove it if it cannot by null. We usually use `Objects.requireNonNull` in constructors not really within the code like this.